src/query/service/src/interpreters/interpreter_merge_into.rs

```diff
@@ -343,6 +367,223 @@ impl MergeIntoInterpreter {
         Ok((physical_plan, table_info.clone()))
     }
 
+    async fn build_static_filter(
+        &self,
+        join: &SExpr,
+        metadata: &MetadataRef,
+        ctx: Arc<QueryContext>,
+    ) -> Result<Box<SExpr>> {
+        // 1. collect statistics from the source side
+        // plan of source table is extended to:
+        //
+        // AggregateFinal(min(source_join_side_expr),max(source_join_side_expr))
+        //        \
+        //     AggregatePartial(min(source_join_side_expr),max(source_join_side_expr))
+        //         \
+        //       EvalScalar(source_join_side_expr)
+        //          \
+        //         SourcePlan
+        let source_plan = join.child(0)?;
+        let join_op = match join.plan() {
+            RelOperator::Join(j) => j,
+            _ => unreachable!(),
+        };
+        if join_op.left_conditions.len() != 1 || join_op.right_conditions.len() != 1 {
+            return Ok(Box::new(join.clone()));
+        }
+        let source_side_expr = &join_op.left_conditions[0];
+        let target_side_expr = &join_op.right_conditions[0];
```

Comment:
why just use only one condition?

Reply:
Yes, let's continue improving this in later PR